### DISCUSSION TITLE
Script to translate text direct in page

### DISCUSSION THREAD

**Discussion by khabib227** [YoshifumiFuyuno/Search-from-Popup-or-ContextMenu#79] [Q&A] [answered]
@YoshifumiFuyuno currently I use a search engine to translate texts with google translator, from "Auto" language to an another language, but this way I need open a new tab to see the result, is possible make a script to translate the selected text direct in the original website, in the original text place? If is not so hard and complex, you can please write the script to me? But no need write the script if is hard to you ok! Thanks.

**Comment by YoshifumiFuyuno** [comment, marked as answer]
Hi.
If you do it simply, you can do it like this.
Please register the engine Method as Bookmarklet.

```javascript
javascript:{
  const sel = getSelection(),
    text = sel.toString().trim(),
    range = sel.rangeCount && sel.getRangeAt(0);

  // https://github.com/thedaviddelta/lingva-translate  auto -> ja
  text && fetch("https://translate.plausibility.cloud/api/v1/auto/ja/" + encodeURIComponent(text))
    .then(response => response.json())
    .then(data => {
      const s = document.createElement("span");
      s.title = text;
      s.textContent = data.translation;
      range.deleteContents();
      range.insertNode(s);
    })
    .catch(alert);
}
```

If you want to do it in earnest, you need to translate each node to support URL links and so on.

P.S. If you search the Internet, you may find better Bookmarklet published.
SPC can also run common Bookmarklets as an engine.

Best regards

**Reply by khabib227** [reply]
Thanks so much the attention!

**Comment by khabib227** [comment]
@YoshifumiFuyuno I'm here again just to say that your script is working very well, is what I wanted, thanks again for help me with your knowledge! And I didn't know about the lingva-translate project, I liked to know about it. :)

**Reply by YoshifumiFuyuno** [reply]
I'm glad I could be of help:smile:
